File: src/main/java/de/tectoast/emolga/commands/admin/ClearCommand.java
package de.tectoast.emolga.commands.admin;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;

import java.util.stream.Collectors;

public class ClearCommand extends Command {
    public ClearCommand() {
        super("clear", "Cleart den Channel", CommandCategory.Flo);
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        e.getChannel().deleteMessages(e.getChannel().getIterableHistory().stream().collect(Collectors.toList())).queue();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/admin/ClearCommand.kt
package de.tectoast.emolga.commands.admin

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import java.util.stream.Collectors

class ClearCommand : Command("clear", "Cleart den Channel", CommandCategory.Flo) {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override fun process(e: GuildCommandEvent) {
        e.textChannel.deleteMessages(e.textChannel.iterableHistory.stream().collect(Collectors.toList())).queue()
    }
}